core/lib/instrument.c

```diff
@@ -5803,6 +5803,10 @@ dr_save_reg(void *drcontext, instrlist_t *ilist, instr_t *where, reg_id_t reg,
 
     CLIENT_ASSERT(reg_is_pointer_sized(reg), "dr_save_reg requires pointer-sized gpr");
 
+#    ifdef AARCH64
```

Comment:
I would test against X86, i.e, #ifndef X86. AFAIR, I think this check also applies for T32 instructions as well.

Reply:
Double checked how drreg does it: https://github.com/DynamoRIO/dynamorio/blob/4964f9e063ef7daea574dfd1b3c2cd10d4b15592/ext/drreg/drreg.c#L827